tests/unit/ui/components/TestShowMoreCard.js

```diff
@@ -1,105 +1,214 @@
 import * as React from 'react';
-import { shallow, mount } from 'enzyme';
+import { shallow } from 'enzyme';
 
 import ShowMoreCard, {
+  extractId,
   ShowMoreCardBase,
   MAX_HEIGHT,
 } from 'ui/components/ShowMoreCard';
 import {
+  applyUIStateChanges,
   createFakeEvent,
   fakeI18n,
   shallowUntilTarget,
 } from 'tests/unit/helpers';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
 
 describe(__filename, () => {
-  const render = (props) => {
+  const getProps = ({ i18n = fakeI18n(), ...props } = {}) => {
+    return {
+      i18n,
+      id: 'showMoreCard',
+      store: dispatchClientMetadata().store,
+      ...props,
+    };
+  };
+
+  function render({ children, ...otherProps } = {}) {
+    const props = getProps(otherProps);
     return shallowUntilTarget(
-      <ShowMoreCard i18n={fakeI18n()} {...props} />,
+      <ShowMoreCard {...props}>{children || 'some info'}</ShowMoreCard>,
       ShowMoreCardBase,
     );
-  };
+  }
 
-  it('reveals more text when clicking "show more" link', () => {
-    const root = render({ children: 'Hello I am description' });
+  it('reveals more text when clicking "read more" link', () => {
+    const { store } = dispatchClientMetadata();
+    const root = render({ store });
 
-    // We have to manually set the expanded flag to false because we don't have
-    // a clientHeight in the tests.
-    root.setState({ expanded: false });
+    // We are simulating the truncate method call.
+    root.instance().truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
 
-    expect(root).not.toHaveClassName('ShowMoreCard--expanded');
+    applyUIStateChanges({ root, store });
 
-    // We have to shallow this prop because we pass a component to it.
-    expect(shallow(root.prop('footerLink')).html()).toContain(
-      '<span class="visually-hidden">Expand to</span> Read more',
-    );
+    expect(root).toHaveProp('footerLink');
+    const footerLink = root.prop('footerLink');
+    const cardLink = shallow(footerLink);
```

Comment:
You could probably do without this `cardLink` const and just have
 `const moreLink = shallow(footerLink).find('.ShowMoreCard-expand-link');`